### ISSUE TITLE
Source Salesforce: Full Sync fails Constantly

### ISSUE THREAD

**Issue by andresbravog** [airbytehq/airbyte#3914]
## Expected Behavior

The first full Sync should succeed without issues.

## Current Behavior

The first sync constantly fails and 3 attempts are made. The logs don't seem to carry any error and the target (BQ in my case) has a bunch of data (can't tell if all of it).

Subsequent syncs seem to execute the Incremental ingestion correctly but they keep on failing and retrying 3 times each day.

<img width="913" alt="Screenshot 2021-06-06 at 21 30 22" src="https://user-images.githubusercontent.com/363321/120937559-6ac99f00-c70e-11eb-9a5b-75f9ca8e8f59.png">

This error has been reproduced by @marcosmarxm and he pointed me to the ticket creation here in Github, please let me know if I can provide any additional detail.

## Logs

[logs-22-0.txt](https://github.com/airbytehq/airbyte/files/6604768/logs-22-0.txt)

## Steps to Reproduce

1. Create a SalesForce Source
2. Create any Target (in my case BQ)
3. Create a connector and start the first Sync

Note how the sync fails even if the logs seems to end correctly and the normalization runs perfectly.

## Severity of the bug for you

Medium

## Airbyte Version

0.24.4-alpha

## Connector Version (if applicable)

0.2.1

## Additional context

??

**Comment by marcosmarxm** [comment]
Thanks @andresbravog! @sherifnada could take a look at this issue? Look data is transfer and normalization works, but some stream are failing.

```
2021-06-03 13:41:08 ERROR (/tmp/workspace/22/0) LineGobbler(voidCall):69 - Traceback (most recent call last):
2021-06-03 13:41:08 ERROR (/tmp/workspace/22/0) LineGobbler(voidCall):69 -   File "/usr/local/bin/base-python", line 8, in <module>
2021-06-03 13:41:08 ERROR (/tmp/workspace/22/0) LineGobbler(voidCall):69 -     sys.exit(main())
2021-06-03 13:41:08 ERROR (/tmp/workspace/22/0) LineGobbler(voidCall):69 -   File "/usr/local/lib/python3.7/site-packages/base_python/entrypoint.py", line 135, in main
2021-06-03 13:41:08 ERROR (/tmp/workspace/22/0) LineGobbler(voidCall):69 -     launch(source, sys.argv[1:])
2021-06-03 13:41:08 ERROR (/tmp/workspace/22/0) LineGobbler(voidCall):69 -   File "/usr/local/lib/python3.7/site-packages/base_python/entrypoint.py", line 120, in launch
2021-06-03 13:41:08 ERROR (/tmp/workspace/22/0) LineGobbler(voidCall):69 -     AirbyteEntrypoint(source).start(args)
2021-06-03 13:41:08 ERROR (/tmp/workspace/22/0) LineGobbler(voidCall):69 -   File "/usr/local/lib/python3.7/site-packages/base_python/entrypoint.py", line 111, in start
2021-06-03 13:41:08 ERROR (/tmp/workspace/22/0) LineGobbler(voidCall):69 -     for message in generator:
2021-06-03 13:41:08 ERROR (/tmp/workspace/22/0) LineGobbler(voidCall):69 -   File "/usr/local/lib/python3.7/site-packages/base_singer/singer_helpers.py", line 167, in read
2021-06-03 13:41:08 ERROR (/tmp/workspace/22/0) LineGobbler(voidCall):69 -     for line, text_wrapper in SingerHelper._read_lines(p):
2021-06-03 13:41:08 ERROR (/tmp/workspace/22/0) LineGobbler(voidCall):69 -   File "/usr/local/lib/python3.7/site-packages/base_singer/singer_helpers.py", line 201, in _read_lines
2021-06-03 13:41:08 ERROR (/tmp/workspace/22/0) LineGobbler(voidCall):69 -     raise Exception(f"Underlying command {process.args} failed with exit code {process.returncode}")
2021-06-03 13:41:08 ERROR (/tmp/workspace/22/0) LineGobbler(voidCall):69 - Exception: Underlying command tap-salesforce --config /tmp/tmprjn4fhni/config.json --properties singer_rendered_catalog.json --state input_state.json failed with exit code 1
```

**Comment by po3na4skld** [comment]
1. Update SAT tests - 1h
2. Run them and fix any errors there - 3h
2. Reproduce the issue - 4h
3. Try to find which streams are failing and why - 8h

**Comment by po3na4skld** [comment]
Blocked by #4564

**Comment by po3na4skld** [comment]
Could not reproduce the issue. I run it and the sync finished successfully. As I hadn't been able to run it with the whole schema file, I did it with the base streams from configured_catalog.json. I think the general problem was due to the #4564 and the large schema size. @sherifnada I attached the screenshot of successful sync below.
 
<img width="887" alt="Screenshot 2021-07-09 at 11 49 54" src="https://user-images.githubusercontent.com/36486349/125051607-c8556080-e0ab-11eb-8ae4-3be4c9bcf2c2.png">
